inst/include/child_model_simulation.hpp

```diff
@@ -13,54 +13,70 @@ void run_child_ageing(int time_step,
                       const State<ModelVariant, real_type> &state_curr,
                       State<ModelVariant, real_type> &state_next,
                       IntermediateData<ModelVariant, real_type> &intermediate) {
-  const auto demog = pars.base.demography;
-  const auto cpars = pars.children.children;
+  const auto survival_probability = pars.base.demography.survival_probability;
+  const auto hc_cd4_transition = pars.children.children.hc_cd4_transition;
+  const auto p_idx_fertility_first = pars.base.options.p_idx_fertility_first;
+
+  const auto hc1_hiv_pop_curr = state_curr.children.hc1_hiv_pop;
+  const auto hc1_art_pop_curr = state_curr.children.hc1_art_pop;
+  const auto hc2_hiv_pop_curr = state_curr.children.hc2_hiv_pop;
+  const auto hc2_art_pop_curr = state_curr.children.hc2_art_pop;
+
+  auto hc1_hiv_pop_next = state_next.children.hc1_hiv_pop;
+  auto hc1_art_pop_next = state_next.children.hc1_art_pop;
+  auto hc2_hiv_pop_next = state_next.children.hc2_hiv_pop;
+  auto hc2_art_pop_next = state_next.children.hc2_art_pop;
```

Comment:
This is fine to me. But also, I think that the previous structure here with shortening `pars.base.demography` -> `demog` and `pars.children.children` -> `cpars` is pretty good and perhaps preferred.

That is in contrast to, for example, the below where `intermediate.base.*****` gets written out a lot and `pars.base.natural_history.*****` gets written out, even though there is a prefix:
https://github.com/mrc-ide/frogger/blob/bd7ee9b00e1de1ed44f9aff28fffc83a4336b046/inst/include/model_simulation.hpp#L56-L76


I might advocate for further standardizing these: e.g. `demog` -> `dempar` and `cpars` -> `chpar` or something like that

And `natural_history.*****` in the adult model, prefer shortening to something like `nh.*****`

(Maybe we standardise them all to 2-letter prefixes and just make some notes about the conventions there)

Reply:
So suggesting Jeff that e.g. instead of the line

```
hc1_hiv_pop_next(hd, cat, a, s) += hc1_hiv_pop_curr(hd, cat, a - 1, s) * survival_probability(a, s, time_step);
```

You would have?

```
cpars.hc1_hiv_pop_next(hd, cat, a, s) += cpars.hc1_hiv_pop_curr(hd, cat, a - 1, s) * demog.survival_probability(a, s, time_step);
```

That is less unpacking! So definitely a fan of that if it still seems readable to you? (I think the first is easier to read)

Reply:
Thanks, I am happy with the first also, it just seemed like a lot of lines of assignment transcribing at the top of the function. 

Would you be open to shortening the prefixes + what do you think, e.g.

```
auto hc = state.children;
auto dp = pars.base.demography;
hc.hc1_hiv_pop_next(hd, cat, a, s) += hc.hc1_hiv_pop_curr(hd, cat, a - 1, s) * dp.survival_probability(a, s, time_step)
```

(I think `hc1_hiv_pop_next` would not belong to `cpars.`)

Reply:
rob and I also had a chat and he agrees that the prefixes are fine! so will be doing that, ill put together a list of proposed prefix names for the meeting tomorrow